modules/dedup/dedup.go

```diff
@@ -28,25 +28,31 @@ import (
 	"k8s.io/klog/v2"
 )
 
-// LeafIdx holds a LeafID and an Idx for deduplication
-type LeafIdx struct {
+// LeafDedupInfo, enables building idempotent deduplicated add-chain responses.
```

Comment:
I think this is all best-effort dedup at the moment, so probably should be careful about implying idempotency here.